code/modules/modular_computers/file_system/programs/atmosscan.dm

```diff
@@ -8,18 +13,72 @@
 	tgui_id = "NtosGasAnalyzer"
 	program_icon = "thermometer-half"
 
+	var/atmozphere_mode = ATMOZPHERE_SCAN_ENV
+	var/list/last_gasmix_data
+
+/// Secondary attack self.
+/datum/computer_file/program/atmosscan/proc/turf_analyze(datum/source, mob/user)
+	SIGNAL_HANDLER
+	if(atmozphere_mode != ATMOZPHERE_SCAN_CLICK)
+		return
+	atmos_scan(user=user, target=get_turf(computer), silent=FALSE)
+	on_analyze(source=source, target=get_turf(computer))
+	return COMPONENT_CANCEL_ATTACK_CHAIN
+
+/// Keep this in sync with it's tool based counterpart [/obj/proc/analyzer_act] and [/atom/proc/tool_act]
+/datum/computer_file/program/atmosscan/tap(atom/A, mob/living/user, params)
+	if(atmozphere_mode != ATMOZPHERE_SCAN_CLICK)
+		return FALSE
+	atmos_scan(user=user, target=A, silent=FALSE)
```

Comment:
can you make this an if and return false if it fails, otherwise it acts like it would successfully atmos scan things it otherwise wouldn't, like objects.